constants.py

```diff
@@ -76,9 +80,13 @@ def _merge_vars(base_vars: JsonType, vars: JsonType) -> None:
 # NOTE: pyinstaller will set this to its own executable when building,
 # detect this to use __file__ and main.py redirection instead
 SELF_PATH = Path(sys.argv[0]).absolute()
+APPIMAGE_PATH = IS_APPIMAGE and Path(os.environ["APPIMAGE"]).absolute()
 if SELF_PATH.stem == "pyinstaller":
     SELF_PATH = Path(__file__).with_name("main.py").absolute()
-WORKING_DIR = SELF_PATH.absolute().parent
+if IS_APPIMAGE:
+    WORKING_DIR = APPIMAGE_PATH.parent
+else:
+    WORKING_DIR = SELF_PATH.parent
```

Comment:
```suggestion
WORKING_DIR = SELF_PATH.parent
```

Reverts the change. `WORKING_DIR` should always be `SELF_PATH.parent`, so that all persistent files (cache, settings.json, etc.) appear next to the main executable / script.